### PR TITLE
Unified and refactor CTAs and links

### LINKED ISSUE #973: ❤️👍👍   Unified and refactor links, ctas, and buttons
We have different ways to "link" to another page:

- Basic links
Screen Shot 2020-10-06 at 13 33 44

- Basic links on dark page:
Screen Shot 2020-10-19 at 15 42 22

- Footer links
Screen Shot 2020-10-06 at 13 35 10

- Ctas = Links with dynamic icon
Screen Shot 2020-10-06 at 13 36 16
Screen Shot 2020-10-06 at 13 36 39

- Landing cta (primary) = Style like a button and always have an arrow (even if it's a external link)
Screen Shot 2020-10-06 at 13 38 26

- Landing cta (secondary) = Style like a button and always have an arrow (even if it's a external link)
Screen Shot 2020-10-06 at 13 39 00

- Ctas with button style
Screen Shot 2020-10-06 at 13 39 38

We should try to refactor these components in order to reuse them properly and to avoid the use of the `!important` CSS property in the last one.

Also, we should clarify with design if the Landings cta should have always an arrow and which elements should have the on hover effect

### PR DESCRIPTION
**Pull request by korgan00**
Fix #973

### CODE REVIEW

**Review by y4izus — CHANGES_REQUESTED** [review]
Thank you, Goyo! I know this is a hard refactor and maybe the review will be also. May the force be with you! 💪  If you need help, just let me know!. 

Let's start the review with the styles:

- When hovering on the primary cta on the home page, the text turns into blue
Screen Shot 2020-11-25 at 15 49 13

- Links on the Quick start section doesn't follow the same styles (I didn't visit the link and its already purple)
Screen Shot 2020-11-25 at 15 50 53

- Same here:
Screen Shot 2020-11-25 at 15 55 43

- There is a moment when hovering the IBM Quantum Experience that the link turns blue but I wasn't able to take the picture. 

- Same for the "Get started" and "See docs" on `/overview`
Screen Shot 2020-11-25 at 15 58 13


- The ghost cta color should be purple and here it's blue
Screen Shot 2020-11-25 at 15 54 19

- Ghost links on learn page, are blue instead of purple
Screen Shot 2020-11-25 at 16 00 13

- Text has disappear from the primary cta to start an event. When  hovering turns into blue and then into purple
Screen Shot 2020-11-25 at 16 04 44

When you finish, please, ask me for a second review

File: components/ui/AppCta.vue [hunk 1]
```diff
@@ -2,31 +2,41 @@
...
 
 @Component
 export default class AppCta extends Vue {
-  get isExternal () {
-    return AppLink.isExternal(this.$attrs.url)
+  @Prop({ type: String, default: 'primary' }) kind!: 'primary'|'secondary'|'ghost'
+  @Prop({ type: String, default: 'light' }) theme!: 'light'|'dark'
+
+  get iconPerLinkType (): string {
+    if (AppLink.isExternal(this.$attrs.url)) {
```
Comment (y4izus) [inline components/ui/AppCta.vue:33]: You're using `this.$attrs.url` two times, could you extract it to a variable?

File: pages/events/index.vue [hunk 2]
```diff
@@ -327,13 +327,6 @@ export default class EventsPage extends QiskitPage {
       margin-top: $layout-02;
       margin-bottom: $layout-03;
     }
-
-    &__cta {
```
Comment (y4izus) [inline pages/events/index.vue:331]: You forgot to remove the [line where this class was used](https://github.com/Qiskit/qiskit.org/pull/1145/files#diff-e89e9da7098c8b828fffda21f9b905c5668d8726f92332ddd3612c55d7bc0d9eR77)

**Review by y4izus — CHANGES_REQUESTED** [review]
Thanks for implementing the first requested chages! Let's go with the second round:

- Please, review all the primary and secondary ctas on small devices, they are all too big:
Screen Shot 2020-11-26 at 11 10 46
Screen Shot 2020-11-26 at 11 11 09
Screen Shot 2020-11-26 at 11 11 21
and so on...

- I also added some comments on the code 👇 

Ask me for another review when you finished! 💪 💪

File: components/landing/TheFeatures/index.vue [hunk 3]
```diff
@@ -37,7 +32,13 @@ import Vue from 'vue'
 import { Component } from 'vue-property-decorator'
 
 @Component
-export default class TheFeatures extends Vue { }
+export default class TheFeatures extends Vue {
+  fullOverviewLink = {
+    url: '/overview',
+    label: 'Full Overview',
```
Comment (techtolentino) [inline components/landing/TheFeatures/index.vue:38]: @korgan00 - please up date this label to `Full overview`

File: components/landing/TheLearnSection/LearnCard.vue [hunk 4]
```diff
@@ -24,7 +22,13 @@ import Vue from 'vue'
 import { Component } from 'vue-property-decorator'
 
 @Component
-export default class LearnCard extends Vue {}
+export default class LearnCard extends Vue {
+  learnLink = {
+    url: 'learn',
+    label: 'Get Learning',
```
Comment (techtolentino) [inline components/landing/TheLearnSection/LearnCard.vue:28]: same type of comment here, please update the label to `Get learning`

**Review by techtolentino — APPROVED** [review]
@korgan00 - great work! I only had two small updates to the labels... 

thank you

**Review by eddybrando — APPROVED** [review]
Awesome job, @korgan00! Thanks a lot!

I really like the great effort put into unifying our buttons and links.

I have just some small suggestions regarding styling _:focus_ similarly to _:hover_ and probably adding a couple of lines more of documentation for future reference. 🙂

**Review by eddybrando — APPROVED** [review]
Awesome, thanks a lot for the changes @korgan00 🙂

File: components/layouts/banners/TheBlackLivesMatterBanner.vue [hunk 5]
```diff
@@ -28,7 +28,11 @@ export default class TheBlackLivesMatterBanner extends Vue {}
     @include type-style('body-short-02');
   }
 
-  &__link {
+  // Rely on BEM specifity because of the problems of css inconsistent ordering on nuxt compilation.
```
Comment (y4izus) [inline components/layouts/banners/TheBlackLivesMatterBanner.vue:31]: Thank you for this reference!

**Review by y4izus — APPROVED** [review]
You did an awesome job @korgan00 !!! thank you very much!!

**Comment by JRussellHuffman** [comment]
Looked over everything. Overall, its great! Noticed a handful of things.
- the background gray on the secondary button should be cool-gray-10 (#F2F4F8). Right now its #f4f4f4
- not sure if the quick start options are a part of this scope, but the text should be cool gray 80
- the downward arrow on the “Get Started” CTA on the overview page looks shifted down by a few pixels by default. It might not be, but wanted to flag it just in case

**Comment by JRussellHuffman** [comment]
created a separate issue for the quick start text color #1206

**Comment by y4izus** [comment]
Fixes #1204

**Pull request merged**